ui/src/components/drafts/DraftEditor.js

```diff
@@ -0,0 +1,147 @@
+import _ from "lodash";
+import React from "react";
+import PropTypes from "prop-types";
+
+import { connect } from "react-redux";
+
+import { Box, Toast } from "grommet";
+
+import {
+  fetchAndAssignSchema,
+  initForm,
+  formDataChange
+} from "../../actions/drafts";
+
+import DepositForm from "./form/Form";
+import DraftActionsHeader from "./components/DraftActionsHeader";
+import Sidebar from "./components/DepositSidebar";
+import DraftJSONPreviewer from "./components/DraftJSONPreviewer";
+
+const transformSchema = schema => {
+  const schemaFieldsToRemove = [
+    "_access",
+    "_deposit",
+    "_cap_status",
+    "_buckets",
+    "_files",
+    "$ana_type",
+    "$schema",
+    "general_title",
+    "_experiment",
+    "control_number"
+  ];
+
+  schema.properties = _.omit(schema.properties, schemaFieldsToRemove);
+  schema = {
+    type: schema.type,
+    properties: schema.properties,
+    dependencies: schema.dependencies
+  };
+
+  return schema;
+};
+
+class CreateDeposit extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  componentDidMount() {
+    // If "schema_id" from URL exists init form and fetch schema
+    if (this.props.match.params.schema_id) {
+      this.props.initForm();
+      this.props.fetchAndAssignSchema(null, this.props.match.params.schema_id);
+    }
+    // else check if schemaId exist on state and fetch schema if needed
+    else if (this.props.schemaId) {
+      // If form schemas are empty, then fetch
+      if (this.props.schemas == null) {
+        this.props.fetchAndAssignSchema(this.props.schemaId);
+      }
+      // If form schemas aren't empty and schemaId different than the one needed
+      // fetch and assign the correct schema
+      else if (this.props.schemaId != this.props.schemas.schemaId) {
+        this.props.fetchAndAssignSchema(this.props.schemaId);
+      }
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.props.schemaId) {
+      if (this.props.schemas == null) {
+        this.props.fetchAndAssignSchema(this.props.schemaId);
+      } else if (this.props.schemaId != this.props.schemas.schemaId) {
+        this.props.fetchAndAssignSchema(this.props.schemaId);
+      }
```

Comment:
if (this.props.schemas == null || this.props.schemaId != this.props.schemas.schemaId) {
  this.props.fetchAndAssignSchema(this.props.schemaId);
}